pandas/core/arrays/datetimelike.py

```diff
@@ -517,8 +517,12 @@ def __getitem__(self, key):
                 return self._box_func(val)
             return type(self)(val, dtype=self.dtype)
 
+        if is_list_like(key) and not isinstance(key, tuple):
+            key = check_array_indexer(self, key)
+
```

Comment:
why is this actually needed? is this not handled by check_array_indexer? if not why not)

Reply:
It'd prefer to keep `check_array_indexer` somewhat strict in what it accepts, so I don't think it should accept / check for tuples. The caller often already knows whether it has a list / array -like or a scalar key (tuple).

Reply:
Not sure which line you are commenting on.

Reply:
referring to L 525, why do you need to call is_bool_indexerif we have already converterted the key at this point.; virtually everywhere else you have remove is_bool_indexer, except here

Reply:
Because here, in `DatetimeArray.__getitem__`, we have special code to check if a boolean mask can be converted into a slice. 
I personally don't like this special case, but removing it is out of scope for this PR for 1.0, so I just made the minimal change to keep existing behaviour here.